src/System.Text.Json/src/System/Text/Json/Node/JsonString.cs

```diff
@@ -208,5 +210,73 @@ public string Value
         ///   Returns <see cref="JsonValueKind.String"/>
         /// </summary>
         public override JsonValueKind ValueKind { get => JsonValueKind.String; }
+
+        /// <summary>
+        ///   Converts the text value of this instance, which should encode binary data as base-64 digits, to an equivalent 8-bit unsigned <see cref="byte"/> array.
+        ///   The return value indicates wether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the <see cref="byte"/> array equivalent of the text contained in this instance,
+        ///   if the conversion succeeded.
+        /// </param>
+        /// <returns>
+        ///   <see langword="true"/> if text was converted successfully; othwerwise returns <see langword="false"/>.
+        /// </returns>
+        internal bool TryGetBytesFromBase64(out byte[] value)
+        {
+            Debug.Assert(_value != null);
+
+            // Shortest length of a base-64 string is 4 characters.
+            if (_value.Length < 4)
+            {
+                value = default;
+                return false;
+            }
+
+#if BUILDING_INBOX_LIBRARY
+            // we decode string -> byte, so the resulting length will
+            // be /4 * 3 - padding. To be on the safe side, keep padding and slice later
+            int bufferSize = _value.Length / 4 * 3;
+
+            byte[] arrayToReturnToPool = null;
+            Span<byte> buffer = bufferSize <= JsonConstants.StackallocThreshold
+                ? stackalloc byte[bufferSize]
```

Comment:
```suggestion
                ? stackalloc byte[JsonConstants.StackallocThreshold]
```

Reply:
@ahsonkhan does https://github.com/dotnet/corefx/pull/38048#discussion_r288921552 still hold?

Reply:
It didn't apply then, either ;)

Reply:
Why would you want to allocate more than what is necessary? Buffersize is guaranteed to be large enough or larger than necessary.

Reply:
I would guess I'm missing something. Maybe the size difference is negligible, and using the constant is part of some convention I'm not familiar with.

Reply:
The JIT can produce better code with stack-allocing a constant-size. It boils down to just a stack-address subtraction.
When stack-allocing `bufferSize` more code will be generated. Also keep in mind, that the JIT rounds up (to a proper size) anyway.
The benefit of a constant is even higher, when local don't need to be initialized with zero.

Reply:
Alright, thanks @gfoidl for the input. I'll commit @stephentoub's suggestion.